### PR TITLE
Add AIFC module stubs

### PR DESCRIPTION
**Pull request by CraftSpider**
Add type stubs for the AIFC module. Should be another checkbox for #1019 and #1147, presuming I have the types right.

### CODE REVIEW

**Comment by srittau** [comment]
Thank you! Without doing a proper review yet, could you remove the empty lines between functions and inside class bodies? We prefer conciseness in typeshed, see our [contribution guidelines](/python/typeshed/blob/master/CONTRIBUTING.md#conventions).

File: stdlib/2and3/aifc.pyi [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+from types import TracebackType
+import sys
+
+class Error(Exception): ...
+
+_aifc_params = NamedTuple('_aifc_params', [('nchannels', int), ('sampwidth', int), ('framerate', int),
+                                           ('nframes', int), ('comptype', bytes), ('compname', bytes)])
+
+_F = Union[Text, IO[bytes]]
+_M = Tuple[int, int, bytes]
```
Comment (JelleZijlstra) [inline stdlib/2and3/aifc.pyi]: Could you give this a more readable name, maybe `_Marker`?

File: stdlib/2and3/aifc.pyi [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    def getparams(self) -> _aifc_params: ...
+    def setmark(self, id: int, pos: int, name: bytes) -> None: ...
+    def getmark(self, id: int) -> _M: ...
+    def getmarkers(self) -> Optional[List[_M]]: ...
+    def tell(self) -> int: ...
+    def writeframesraw(self, data: Any) -> None: ...  # Actual type for data is Buffer Protocol
+    def writeframes(self, data: Any) -> None: ...
+    def close(self) -> None: ...
+
+def open(f: _F, mode: Text = ...) -> Union[Aifc_read, Aifc_write]: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/aifc.pyi]: You can use overloads on Literal to return a more specific return type. Looks like it just supports r, rb, w, and wb.

Also, if f is a file object mode can be None.

Comment (CraftSpider) [reply stdlib/2and3/aifc.pyi]: Added Literal overloads, not many other Literal references currently in the repo but I followed the other example I found as best I could.

File: stdlib/2and3/aifc.pyi [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    def getmarkers(self) -> Optional[List[_Marker]]: ...
+    def tell(self) -> int: ...
+    def writeframesraw(self, data: Any) -> None: ...  # Actual type for data is Buffer Protocol
+    def writeframes(self, data: Any) -> None: ...
+    def close(self) -> None: ...
+
+@overload
+def open(f: _File, mode: Literal["r", "rb"] = ...) -> Aifc_read: ...
+@overload
+def open(f: _File, mode: Literal["w", "wb"]) -> Aifc_write: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/aifc.pyi]: You should also add an overload with `mode: str` that returns `Any`, for cases where the argument is not a literal.

**Comment by CraftSpider** [comment]
All requested changes have been made so far, ready for review again.

File: stdlib/2and3/aifc.pyi [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    def getnframes(self) -> int: ...
+    def getsampwidth(self) -> int: ...
+    def getframerate(self) -> int: ...
+    def getcomptype(self) -> bytes: ...
+    def getcompname(self) -> bytes: ...
+    def getparams(self) -> _aifc_params: ...
+    def getmarkers(self) -> Optional[List[_Marker]]: ...
+    def getmark(self, id: int) -> _Marker: ...
+    def setpos(self, pos: int) -> None: ...
+    def readframes(self, nframes: int) -> str: ...
```
Comment (srittau) [inline stdlib/2and3/aifc.pyi]: Shouldn't this return `bytes`? At least that's what return when `nframes == 0` and it would make more sense to me.

Comment (CraftSpider) [reply stdlib/2and3/aifc.pyi]: `bytes` does seem to make more sense to me. It has been long enough that I'm not sure my logic in making it a `str`

**Comment by srittau** [comment]
Just waiting for @JelleZijlstra's approval now.

**Pull request merged**